acceptance/assertions/output.go

```diff
@@ -0,0 +1,276 @@
+}
+
+func (o OutputAssertionManager) ReportsCacheCreation(layer string) {
+	o.testObject.Helper()
+	o.testObject.Log("cacher adds layers")
+
+	o.assert.Matches(o.output, regexp.MustCompile(fmt.Sprintf(`(?i)Adding cache layer '%s'`, layer)))
+}
+
+func (o OutputAssertionManager) ReportsSkippingRestore() {
+	o.testObject.Helper()
+	o.testObject.Log("skips restore")
+
+	o.assert.Contains(o.output, "Skipping 'restore' due to clearing cache")
+}
+
+func (o OutputAssertionManager) ReportsSkippingBuildpackLayerAnalysis() {
+	o.testObject.Helper()
+	o.testObject.Log("skips buildpack layer analysis")
+
+	o.assert.Matches(o.output, regexp.MustCompile(`(?i)Skipping buildpack layer analysis`))
+}
+
+func (o OutputAssertionManager) ReportsReadingFileContents(path, expectedContent, phase string) {
+	o.testObject.Helper()
+
+	o.assert.ContainsF(o.output, "%s: Reading file '/platform%s': %s", phase, path, expectedContent)
+}
+
+func (o OutputAssertionManager) ReportsRunImageStackNotMatchingBuilder(runImageStack, builderStack string) {
+	o.testObject.Helper()
+
+	o.assert.Contains(
+		o.output,
+		fmt.Sprintf("run-image stack id '%s' does not match builder stack '%s'", runImageStack, builderStack),
+	)
+}
+
+func (o OutputAssertionManager) WithoutColors() {
+	o.testObject.Helper()
+	o.testObject.Log("has no color")
+
+	o.assert.NoMatches(o.output, regexp.MustCompile(`\x1b\[[0-9;]*m`))
+}
+
+func (o OutputAssertionManager) ReportsBuildStep(message string) {
+	o.testObject.Helper()
+
+	o.assert.ContainsF(o.output, "Build: %s", message)
+}
+
+func (o OutputAssertionManager) ReportsAddingBuildpack(name, version string) {
+	o.testObject.Helper()
+
+	o.assert.ContainsF(o.output, "Adding buildpack '%s' version '%s' to builder", name, version)
+}
+
+func (o OutputAssertionManager) ReportsPullingImage(image string) {
+	o.testObject.Helper()
+
+	o.assert.ContainsF(o.output, "Pulling image '%s'", image)
+}
+
+func (o OutputAssertionManager) ReportsImageNotExistingOnDaemon(image string) {
+	o.testObject.Helper()
+
+	o.assert.ContainsF(o.output, "image '%s' does not exist on the daemon", image)
+}
+
+func (o OutputAssertionManager) ReportsPackageCreation(name string) {
+	o.testObject.Helper()
+
+	o.assert.ContainsF(o.output, "Successfully created package '%s'", name)
+}
+
+func (o OutputAssertionManager) ReportsPackagePublished(name string) {
+	o.testObject.Helper()
+
+	o.assert.ContainsF(o.output, "Successfully published package '%s'", name)
+}
+
+func (o OutputAssertionManager) ReportsCommandUnknown(command string) {
+	o.testObject.Helper()
+
+	o.assert.ContainsF(o.output, `unknown command "%s" for "pack"`, command)
+}
+
+func (o OutputAssertionManager) ReportsConnectedToInternet() {
+	o.testObject.Helper()
+
+	o.assert.Contains(o.output, "RESULT: Connected to the internet")
+}
+
+func (o OutputAssertionManager) ReportsDisconnectedFromInternet() {
+	o.testObject.Helper()
+
+	o.assert.Contains(o.output, "RESULT: Disconnected from the internet")
```

Comment:
It feels like this "manager" is a little overloaded. I can see the value in common pack specific output given that is what the project repo is for. 

On the other hand, abstracting other output that is more specific to a particular test makes it just an abstraction leading to a burden with little to no value if we were to expect the pattern to be followed.

Reply:
Given the `LifecycleOutputAssertionManager` idea floated in a related comment, do you think something like a `TestAppOutputAssertionManager` be a worthwhile abstraction, that would ensure the separation of concerns?

Reply:
Or more accurately `TestBuildpackOutputAssertionManager`... becoming somewhat of a mouthful

Reply:
Hahaha. Yeah, `NewTestBuildpackOutputAssertionManager` is so Java-ey. I think the use of packages may help.

Naming concerns aside, it's still debatable. Without working in it first hand with these set of change I can't say that I have too strong of an opinion. I'm just wondering whether it's reasonable to expect the creation of a new method solely to call another singleline when you could have just called `assert.Contains(...)`. Specially when that method is only used once.

Either way, looking at the latest changes. I'm okay with merging this in and iterating on it if need be.

Reply:
Perfect